stratum/lib/security/credentials_manager.cc

```diff
@@ -13,26 +13,110 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-
 #include "stratum/lib/security/credentials_manager.h"
 
+#include <memory>
+
+#include "absl/memory/memory.h"
 #include "gflags/gflags.h"
+#include "grpcpp/security/server_credentials.h"
+#include "grpcpp/security/server_credentials_impl.h"
+#include "grpcpp/security/tls_credentials_options.h"
 #include "stratum/glue/logging.h"
-#include "absl/memory/memory.h"
+#include "stratum/glue/status/status.h"
+#include "stratum/lib/macros.h"
+#include "stratum/lib/utils.h"
 
-namespace stratum {
+DEFINE_string(ca_cert, "", "CA certificate path");
+DEFINE_string(server_key, "", "gRPC Server pricate key path");
+DEFINE_string(server_cert, "", "gRPC Server certificate path");
 
-CredentialsManager::CredentialsManager() {}
+namespace stratum {
 
 CredentialsManager::~CredentialsManager() {}
 
+CredentialsManager::CredentialsManager() : reload_credential(true) {
+  if (FLAGS_ca_cert.empty() || FLAGS_server_key.empty() ||
+      FLAGS_server_cert.empty()) {
+    LOG(WARNING) << "Using insecure server credentials";
+    server_credentials_ = ::grpc::InsecureServerCredentials();
+  } else {
+    // Load default credentials
+    ::util::Status status;
+    status.Update(::stratum::ReadFileToString(FLAGS_ca_cert, &pem_root_certs_));
+    status.Update(
+        ::stratum::ReadFileToString(FLAGS_server_key, &server_private_key_));
+    status.Update(
+        ::stratum::ReadFileToString(FLAGS_server_cert, &server_cert_));
+    if (!status.ok()) {
+      LOG(WARNING)
+          << "Unable to load credentials, using insecure server credentials";
+      server_credentials_ = ::grpc::InsecureServerCredentials();
+      return;
+    }
+    auto credential_reload_manager_ =
+        std::shared_ptr<TlsCredentialReloadInterface>(
+            static_cast<TlsCredentialReloadInterface *>(this));
```

Comment:
This might not be a good idea, maybe I should still need to create another class and access that class from CredentialManager